lib/scope.js

```diff
@@ -82,6 +77,11 @@ function isStrictScope(scope, block, isMethodDefinition, useDirective) {
         return false;
     }
 
+    // A function with non-block body, like `() => 1`
+    if (!body.body) {
```

Comment:
I'd like to move this check into the `if` block about `scope.type === "function"` above. Because the type of `body` is unknown at this location.

Maybe:

```js
    if (scope.type === "function") {
        if (block.type === Syntax.ArrowFunctionExpression && block.body.type !== Syntax.BlockStatement) {
            return false;
        }
        //....
```

Reply:
I applied this verbatim, eslint-scope and eslint tests still pass. :ok_hand: